Take advantage of simplified logging

knoepfel opened the pull request:
Follow-on to https://github.com/HEPCloud/decisionengine/pull/515.

pep8speaks commented:
Hello @knoepfel! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:











* In the file [`src/decisionengine_modules/AWS/sources/AWSSpotPrice.py`](https://github.com/HEPCloud/decisionengine_modules/blob/fc3d676a93fc4843ca64bdbb67cc6de17f6ec404/src/decisionengine_modules/AWS/sources/AWSSpotPrice.py):

> [Line 5:1](https://github.com/HEPCloud/decisionengine_modules/blob/fc3d676a93fc4843ca64bdbb67cc6de17f6ec404/src/decisionengine_modules/AWS/sources/AWSSpotPrice.py#L5): [F401](https://duckduckgo.com/?q=pep8%20F401) 'structlog' imported but unused
> [Line 14:1](https://github.com/HEPCloud/decisionengine_modules/blob/fc3d676a93fc4843ca64bdbb67cc6de17f6ec404/src/decisionengine_modules/AWS/sources/AWSSpotPrice.py#L14): [F401](https://duckduckgo.com/?q=pep8%20F401) 'decisionengine.framework.modules.logging_configDict.CHANNELLOGGERNAME' imported but unused
> [Line 186:26](https://github.com/HEPCloud/decisionengine_modules/blob/fc3d676a93fc4843ca64bdbb67cc6de17f6ec404/src/decisionengine_modules/AWS/sources/AWSSpotPrice.py#L186): [F821](https://duckduckgo.com/?q=pep8%20F821) undefined name 'config'

* In the file [`src/decisionengine_modules/AWS/sources/AWSSpotPriceWithSourceProxy.py`](https://github.com/HEPCloud/decisionengine_modules/blob/fc3d676a93fc4843ca64bdbb67cc6de17f6ec404/src/decisionengine_modules/AWS/sources/AWSSpotPriceWithSourceProxy.py):

> [Line 5:1](https://github.com/HEPCloud/decisionengine_modules/blob/fc3d676a93fc4843ca64bdbb67cc6de17f6ec404/src/decisionengine_modules/AWS/sources/AWSSpotPriceWithSourceProxy.py#L5): [F401](https://duckduckgo.com/?q=pep8%20F401) 'structlog' imported but unused
> [Line 12:1](https://github.com/HEPCloud/decisionengine_modules/blob/fc3d676a93fc4843ca64bdbb67cc6de17f6ec404/src/decisionengine_modules/AWS/sources/AWSSpotPriceWithSourceProxy.py#L12): [F401](https://duckduckgo.com/?q=pep8%20F401) 'decisionengine.framework.modules.logging_configDict.CHANNELLOGGERNAME' imported but unused
> [Line 132:13](https://github.com/HEPCloud/decisionengine_modules/blob/fc3d676a93fc4843ca64bdbb67cc6de17f6ec404/src/decisionengine_modules/AWS/sources/AWSSpotPriceWithSourceProxy.py#L132): [F821](https://duckduckgo.com/?q=pep8%20F821) undefined name 'logger'





* In the file [`src/decisionengine_modules/AWS/sources/LocalCapacity.py`](https://github.com/HEPCloud/decisionengine_modules/blob/fc3d676a93fc4843ca64bdbb67cc6de17f6ec404/src/decisionengine_modules/AWS/sources/LocalCapacity.py):

> [Line 7:26](https://github.com/HEPCloud/decisionengine_modules/blob/fc3d676a93fc4843ca64bdbb67cc6de17f6ec404/src/decisionengine_modules/AWS/sources/LocalCapacity.py#L7): [F821](https://duckduckgo.com/?q=pep8%20F821) undefined name 'config'























* In the file [`src/decisionengine_modules/NERSC/transforms/CompareNerscUserpoolSlots.py`](https://github.com/HEPCloud/decisionengine_modules/blob/fc3d676a93fc4843ca64bdbb67cc6de17f6ec404/src/decisionengine_modules/NERSC/transforms/CompareNerscUserpoolSlots.py):

> [Line 24:26](https://github.com/HEPCloud/decisionengine_modules/blob/fc3d676a93fc4843ca64bdbb67cc6de17f6ec404/src/decisionengine_modules/NERSC/transforms/CompareNerscUserpoolSlots.py#L24): [F821](https://duckduckgo.com/?q=pep8%20F821) undefined name 'config'















* In the file [`src/decisionengine_modules/glideinwms/transforms/job_clustering.py`](https://github.com/HEPCloud/decisionengine_modules/blob/fc3d676a93fc4843ca64bdbb67cc6de17f6ec404/src/decisionengine_modules/glideinwms/transforms/job_clustering.py):

> [Line 18:9](https://github.com/HEPCloud/decisionengine_modules/blob/fc3d676a93fc4843ca64bdbb67cc6de17f6ec404/src/decisionengine_modules/glideinwms/transforms/job_clustering.py#L18): [F821](https://duckduckgo.com/?q=pep8%20F821) undefined name 'logger'

jcpunk commented:
Can you rebase off `HEAD`?

BrunoCoimbra commented:
Can we also use the new logging boilerplate model for Sources and Publishers as well?
e.g.: `FactoryClientManifests` and `JobClusteringPublisher`

knoepfel commented:
Good catch, @BrunoCoimbra.  I'll update the PR.

knoepfel commented:
@BrunoCoimbra, I adjusted the logging according to your suggestions.  I also adjusted various utility functions so that a logger is "passed in" to the function if logging is desired--this way, we associate the messages from the utility function with the module that uses that utility.

jcpunk commented on src/decisionengine_modules/glideinwms/resource_dist_plugins.py:
```diff
@@ -14,31 +8,35 @@
     for rss, column_name in _RESOURCE_FROM_COLUMN_MAP.items():
         fom_df = resources.get(rss)
-        logger.info(f"Ordering resources based on {rss}")
+        if logger:
```
I'm wondering if `if logger is not None:` is more explicit about what is going on here.  I go back and forth on the `__bool__` methods, but I suspect we care more if a logger is passed, not if it "is true"....

knoepfel replied:
A fair point--changes made accordingly.
